server/monitor/src/main/java/org/apache/accumulo/monitor/Monitor.java

```diff
@@ -221,6 +229,30 @@ long calculateCount() {
     }
   }
 
+  public Query fetchQuery(String tableName, Range range) {
+    if (tableName == null)
+      tableName = "test_ingest";
+    ServerContext context = getContext();
+    Query result = new Query();
+    long count = 0L;
+    log.info("MIKE query value = " + range);
+
+    try {
+      BatchScanner scanner = context.createBatchScanner(tableName, Authorizations.EMPTY);
+      ArrayList<Range> ranges = new ArrayList<>();
+      ranges.add(range);
+      scanner.setRanges(ranges);
+      for (Entry<Key, Value> entry : scanner) {
```

Comment:
This has the potential to bring a lot of data back to the monitor, which could take a really long time to complete.

Reply:
That could kill the monitor, making it unavailable for other users, too. And, unauthenticated users being able to initiate this from a web page would make it a DoS vulnerability.

Reply:
Yeah good point. We could force the user to put in a valid range or something else to prevent full table scan.

Reply:
The monitor generally runs with a low memory footprint, with the purpose of serving multiple users. A single large row could kill it, not just full table scans. Avoiding a full table scan isn't sufficient.

We currently mitigate the amount of load a user can trigger the monitor to perform by limiting our refresh of the stats in the monitor to no more than once every 5 seconds. But, there's no way to limit the load a user could initiate when they can initiate arbitrary work.

Reply:
> The monitor generally runs with a low memory footprint, with the purpose of serving multiple users. A single large row could kill it, not just full table scans. Avoiding a full table scan isn't sufficient.
> 
> We currently mitigate the amount of load a user can trigger the monitor to perform by limiting our refresh of the stats in the monitor to no more than once every 5 seconds. But, there's no way to limit the load a user could initiate when they can initiate arbitrary work.

True.  This would allow them to put load on the cluster.

Reply:
I was thinking maybe we could do a simple user name and password and connect like in the shell but this would save admins time having to ssh to a server, connect to the shell and figure out the shell commands.

Reply:
If we do that, then we'd have to add code to manage sessions, and what about Kerberos or other pluggable authentication stuff? And, the web service increases the surface area for exposing user credentials. It just seems like too much... and not what the monitor was intended to be.